ui-v2/app/mixins/with-event-source.js

```diff
@@ -33,7 +44,10 @@ export default Mixin.create(WithListeners, {
           // setProperties will be called the next time we enter the Route so this
           // is ok for what we need and means that the above conditional works
           // as expected (see 'here' comment above)
-          delete this[prop];
+          // delete this[prop];
+          // TODO: Check that nulling this out instead of deleting is fine
```

Comment:
Is this TODO for this PR or for the future?

Reply:
Ah yeah I meant to ask about this. I'm sure I remember seeing an ember `unset` or similar, but I can't find anything (so something that deletes a property entirely not just sets it to null). Does that ring any bells with you?

I think I tried doing `set(this, prop, undefined)` here also, maybe thats the 'sanctioned' way of completely unsetting a property? In the end I settled with `null`, as far as I remember when I looked where we were relying on the `undefined` check, a `null` check instead seemed just as good.

If you know what the sanctioned way to completely unset a property is in a 3.8 world, that would be really good to know.

I'm gonna relook at the code surrounding this in a sec, just to confirm what I've done here is 'fine' (or have another look at doing a real unset, which I'd prefer), then the TODO will go in this PR.

Reply:
I think `delete` is a fine way of accomplishing this. I don’t think this has changed in Ember in years, at least. You can `set` to `undefined` too, it might have some slight differences in behaviour depending on the context… I guess the thing to do is see how it actually acts 😯

Reply:
Yeah its weird, I couldn't remember the exact error doing this fixes, so I checked back in the commits. Took me a while to find it again but its this:

<img width="735" alt="Screenshot 2019-08-21 at 16 44 24" src="https://user-images.githubusercontent.com/554604/63447172-7661ae00-c433-11e9-9bcd-313bbc27d149.png">

^ that's what happens when you just use vanilla `delete` here. As soon as I change it to use `set`, setting it to `null` or `undefined` I don't get the error anymore. The error message is a little weird and sounds unrelated, but changing the `delete` to something `set` based solves it - and as this is all proxy related it probably makes sense that this fix and the error message are related.

I'm quibbling about using `undefined`, it feels a bit wrong setting things to `undefined`, which is why I chose `null` here instead. When I re-read the code it still achieves the same thing with `null` so I left it at that.

I 'think' I'm gonna leave this as is and remove the TODO, I've had a good look into it and I think this is the best I'm gonna get here. Unless you can think of anything else we can try?

Reply:
Setting to `undefined` is more equivalent to using `delete`, but if it works the same way with either, it doesn’t really matter to me!

Reply:
Cool thanks @backspace I went with `undefined` in the end. The aim here is to change as little as possible aprt from the upgrade, and you're right, setting to `undefined` is closer to a delete.

I also added some more detailed notes here just incase.

Of further note - this is part of the code that will probably go once we add the work we've been doing on our `<DataSource />` component, as none of it will be required anymore - so as long as it gets us over the line with the upgrade, and we can start using `<DataSource />` its not a biggie in the scheme of things.

Thanks for the help and convo!